core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableRelFactories.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.adapter.enumerable;
+
+import org.apache.calcite.rel.RelCollation;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.core.RelFactories;
+import org.apache.calcite.rex.RexNode;
+
+/**
+ * Contains factory interface and default implementation for creating various
+ * rel nodes.
+ */
+public class EnumerableRelFactories {
+
+  public static final org.apache.calcite.rel.core.RelFactories.SortFactory
+      ENUMERABLE_SORT_FACTORY = new SortFactoryImpl();
+
```

Comment:
Why we only have one physical factory ? What about the others ?

Reply:
For this particular change, addressing CALCITE-2970, we only need EnumerableSort. We will add more in the future.

Reply:
Now I add a few more factories for the purpose of UT.